Adding gen level subjets to forest

mverwe opened the pull request:
Describe the Pull-Request:
- added 'groom' option to makeJetSequences: SoftDrop and Filter is not part of background subtraction. Now we also create the jet sequences transparently for pp
- added SoftDrop and Filter to standard pp foresting
- added filtering of signal particles to feed to subjet producers (important for offical embedded samples)
- adding sequences to cluster gen-level subjets using SoftDrop
- modified HiInclusiveJetAnalyzer to store gen level subjet information
- on my way also fixed bug in runtest.sh (we were overwriting the output of MIX by MB. Now to separate root files are produced when running in verbose mode)

Still to do: turn off NSubjettiness module for calo pp jets. It produces a lot of warnings because the constituents of calo jets are not stored in the pp AOD (calo towers). This module is added in the template and I don't know how to not execute it for pp in a transparent way.

I apologize for the many commits. It took a while before I got all the jet sequences correct. I did run many tests to make sure I'm not destroying anything of the existing functionality.
Let me know in case I overlooked something or if you have questions.

Does the PR pass the test script located at
HeavyIonsAnalysis/JetAnalysis/test/runtest.sh
[X] Yes
[] No

Please make sure to mention (using the @ syntax) anyone who might be interested in this PR.

mverwe committed:
add gen-level subjets

mverwe committed:
gen level SoftDrop

mverwe committed:
add gen subjets to HiInclusiveAnalyzer

mverwe committed:
separate sub and groom in makeJetSequences

mverwe committed:
updated jet sequences

mverwe committed:
update jet sequences

mverwe committed:
add gen level subjets to pp running

mverwe committed:
filter subid=0 particles for subjet clustering at gen level

mverwe committed:
fix bug in runtest macro

mverwe committed:
add gen level subjet info to jet tree

mverwe committed:
add comment about calo jets in pp

mverwe committed:
update all jet sequences (again)

mverwe committed:
filter gen particles from genParticlesForJets

mverwe committed:
first select signal particles, then filter particles for jets

mverwe committed:
set R0 for gen-level SoftDrop

mverwe commented:
@richard-cms : is it ok with you if I merge this in?

harmonicoscillator commented:
@mverwe I think it would be good to test the JEC settings since this changes so much. You can test them by changing these lines:
https://github.com/CmsHI/cmssw/blob/forest_CMSSW_7_5_8_patch3/HeavyIonsAnalysis/JetAnalysis/test/runForestAOD_PbPb_MIX_75X.py#L82
(Comment line 83, uncomment line 85 and then do ./runtest.sh again)

If that passes for you I think it's okay to merge if you are happy.

mverwe committed:
signal particle filtering already done in gen sequences

mverwe committed:
fix jec mode for pp mc

mverwe commented:
@richard-cms : thanks for the reminder. I always forget about that sequence which is not tested automatically. I tested now and for PbPb MIX everything was fine. I however also tested the PP JEC mode and noticed that all kind of calo jets were included which we cannot run because we don't have towers in the event. So I adjusted those configs to make it run again. (last commit) @dgulhan 

@yetkinyilmaz : I moved the signal particle filter to the gen sequences. Currently it was being loaded twice. Now it is together with the filter of particles for jets.
